src/main/java/mate/academy/rickandmorty/controller/CharacterController.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.rickandmorty.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.internal.CharacterSearchParam;
+import mate.academy.rickandmorty.model.CharacterEntity;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/character")
+@Tag(name = "Character API", description = "API endpoints for managing characters.")
+public class CharacterController {
+    private final CharacterService characterService;
+
+    @GetMapping("/random")
+    @Operation(summary = "Get a Random CharacterEntity",
+            description = "Retrieve a random characterEntity from the database.")
+    public CharacterEntity getRandomCharacter() {
+        return characterService.getRandomCharacter();
+    }
+
+    @GetMapping("/search")
```

Comment:
Can't see necessity of url suffix, simple filtered GET search is usually on same url as getAll operation but with arguments
structure with /search suffix would be preferred if we have complex conditions and wish to switch to POST type of request with some request body
```suggestion
    @GetMapping
```